### ISSUE TITLE
Obtain Mail.ru OAuth token from browser

### ISSUE THREAD

**Issue by ivandeex** [ivandeex/rclone#7]
The backend configuration should run a browser for user if the configured password is an empty string. This will prepare the backend for future SMS authentication announced by mailru.

**Comment by ivandeex** [comment]
According to Mailru documentation at https://api.mail.ru/docs/guides/oauth/standalone/ developers should register their application at https://api.mail.ru/apps/my/add and at the next step only IFRAME applications are allowed.

Then, if we register an application and send a request like https://connect.mail.ru/oauth/authorize?client_id=MY_APP_ID&response_type=token&redirect_uri=http%3A%2F%2F127.0.0.1 the response will not be redirected to localhost but rather to http://connect.mail.ru/oauth/success.html#REPLY i.e. Mailru insists that some javascript (e.g. from IFRAME) handles the redirect and decodes the REPLY.

The official Disk-O: application uses either username/password or non-token authorization based on S256 challenge (see here https://tools.ietf.org/html/rfc7636).

A typical S256 request from `Disk-O:` looks like https://oauth.mail.ru/login?scope=userinfo+cloud.api&client_id=f2e52f9829da40be846cdc87376eafb5&redirect_uri=http%3A%2F%2Flocalhost%3A4546%2Foauth%2Fmailru&code_challenge_method=S256&code_challenge=xxx&response_type=code&state=xxx

and redirects to http://localhost:4546/oauth/mailru?code=xxx&state=xxx
I doubt one can easily get input data (secret keys) from Mailru to implement this in an external app.

**Bottom line:
It is not possible to integrate rclone with official way of Mailru authentication.
Stay with token authorization based on user/password. It works fine and won't break in near future.**
